done/done.py

```diff
@@ -1,77 +1,110 @@
 from xblock.fragment import Fragment
 
+try:
+    from eventtracking import tracker
+except ImportError:
+    class tracker(object):  # pylint: disable=invalid-name
+        """
+        Define tracker if eventtracking cannot be imported. This is a
+        workaround so that the code works in both edx-platform and
+        XBlock workbench (the latter of which does not support event
+        emission). This should be replaced with XBlock's emit(), but
+        at present, emit() is broken.
+        """
+        def __init__(self):
+            """ Do nothing """
+            pass
+
+        @staticmethod
+        def emit(param1, param2):
+            """ In workbench, do nothing for event emission """
+            pass
+
+
+def resource_string(path):
+    """Handy helper for getting resources from our kit."""
+    data = pkg_resources.resource_string(__name__, path)
+    return data.decode("utf8")
+
 
 class DoneXBlock(XBlock):
     """
     Show a toggle which lets students mark things as done.
     """
 
-    # Fields are defined on the class.  You can access them in your code as
-    # self.<fieldname>.
     done = Boolean(
-           scope = Scope.user_state, 
-           help = "Is the student done?",
-           default = False
-        )
+        scope=Scope.user_state,
+        help="Is the student done?",
+        default=False
+    )
 
     align = String(
-           scope = Scope.settings, 
-           help = "Align left/right/center",
-           default = "left"
-        )
+        scope=Scope.settings,
+        help="Align left/right/center",
+        default="left"
+    )
 
     has_score = True
 
-    def resource_string(self, path):
-        """Handy helper for getting resources from our kit."""
-        data = pkg_resources.resource_string(__name__, path)
-        return data.decode("utf8")
-
+    # pylint: disable=unused-argument
     @XBlock.json_handler
     def toggle_button(self, data, suffix=''):
-        self.done = data['done']
-        if data['done']:
-            grade = 1
-        else:
-            grade = 0
-
-        self.runtime.publish(self, 'grade', {'value':grade, 'max_value': 1})
-        return {}
-
-
-    def student_view(self, context=None):
+        """
+        Ajax call when the button is clicked. Input is a JSON dictionary
+        with one boolean field: `done`. This will save this in the
+        XBlock field, and then issue an appropriate grade.
+        """
+        if 'done' in data:
+            self.done = data['done']
+            if data['done']:
+                grade = 1
+            else:
+                grade = 0
+            grade_event = {'value': grade, 'max_value': 1}
+            self.runtime.publish(self, 'grade', grade_event)
+            # This should move to self.runtime.publish, once that pipeline
+            # is finished for XBlocks.
+            tracker.emit("edx.done.toggle", {'done': self.done})        
```

Comment:
Thanks for tagging me @mulby . I agree that it would be better to distinguish the client and server side events further to have different names. 

1. @pmitros , Could I ask whether the client and server events are semantically different -- e.g. is the client event a 'toggle.requested', while the server event is a 'toggle.completed'?  We can chat in person here if you'd like.

2. By the way, Piotr, to help ensure consistency across features, we've instituted an [event design & docs review process](https://openedx.atlassian.net/wiki/display/AN/Event+Design+and+Review+Process). This surfaces each new event for review, captures event meaning/format/design choices, educates other teams in designing events, and helps the docs team get them into the research guide.  The first step would be to create an Event Design wiki page by filling out a template, as described here:
https://openedx.atlassian.net/wiki/display/AN/Event+Designs 

This should hopefully not take too long.Then just get a quick Analytics review of that page (me or Victor), and let Docs (Alison) know that this new event is coming to make sure it gets documented in the research guide. I'll also try to find you in person to chat and/or answer any questions you may have.

Thank you in advance -- these design docs really help both the docs team and future community contributors when they are designing their own events.